### PR TITLE
[semantic-release] add Branch and improve Logger typings

### PR DESCRIPTION
**Pull request by GabeDuarteM**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [branch](https://github.com/semantic-release/semantic-release/blob/master/docs/developer-guide/plugin.md#verifyconditions) and [logger](https://github.com/klaussinani/signale/blob/master/types/signale.d.ts#L105) (semantic release uses signale for logging, so I just took their type definitions
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

PS.: Prettier styling was a bit out of date, so the first commit just runs prettier through the typings

### CODE REVIEW

File: types/semantic-release/index.d.ts [hunk 1]
```diff
@@ -422,8 +425,23 @@ declare namespace SemanticRelease {
          * The shared logger instance of semantic release.
          */
         logger: {
-            log: (message: string, ...vars: any[]) => void,
-            error: (message: string, ...vars: any[]) => void,
+            await: LoggerFunction;
```
Comment (peterblazejewicz) [inline types/semantic-release/index.d.ts:428]: where all those came from? it looks logger api is little simpler:
https://github.com/semantic-release/semantic-release/blob/58a226f29c04ee56bbb02cc661f020d568849cad/test/index.test.js#L36-L40

Comment (GabrielDuarteM) [reply types/semantic-release/index.d.ts:428]: semantic release uses signale to do the logging as you can see [here](https://github.com/semantic-release/semantic-release/blob/master/lib/get-logger.js). Signale provides their own typings, so I just took it from [their repo](https://github.com/klaussinani/signale/blob/master/types/signale.d.ts#L105)

You can see below all methods added on the MR when I console.log the logger object
image

The simpler portion you saw is just what they decided to test I imagine (and probably only what they use internally as well)

Comment (peterblazejewicz) [reply types/semantic-release/index.d.ts:428]: ok, sorry. They do not have much docs (which should indicate what should be used)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@GabrielDuarteM thanks!

**Comment by GabrielDuarteM** [comment]
Ready to merge

**Pull request merged**